src/constructs/core/stack.ts

```diff
@@ -120,4 +122,26 @@ export class GuStack extends Stack implements StackStageIdentity, GuMigratingSta
     this.addTag("Stack", this.stack);
     this.addTag("Stage", this.stage);
   }
+
+  protected prepare(): void {
```

Comment:
The [docs](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Construct.html#protected-prepare) are a little scary, but we're only logging so should be ok?!

> Perform final modifications before synthesis.
>
> This method can be implemented by derived constructs in order to perform final changes before synthesis. prepare() will be called after child constructs have been prepared.
>
> This is an advanced framework feature. Only use this if you understand the implications.